### DISCUSSION TITLE
Embind ignore my bindings during linking

### DISCUSSION THREAD

**Discussion by AlexandrMatveev** [emscripten-core/emscripten#19720] [Q&A]
Hey guys! Try to use emcmake + emmake to build my project with [CMakeLists.txt (pastebin)](https://pastebin.com/TbrJwh3b) 
Process is successfully completed but when i try to find exported function in .js part, I don't see any
Only common embind runtime code without my bindings
Maybe anyone can suggest what is wrong with my cmake build script and how i can fix it. Thanks)

**Comment by sbc100** [comment]
Do the object files that contains your `EMSCRIPTEN_BINDINGS` code live inside of library archive (`.a` files)?  If so it could be that the linker is not including them.   Individual object files are only included if your program depends on a symbol with that object file. 

If this is the issue where are several workarounds:

1. Use `-Wl,--whole-archive`  and `-Wl,--no-whole-archive` around the libraries in question (not recommended)
2. Use `-Wl,-u<symbol>` to explicitly reference a symbol within the object file (`-sEXPORTED_FUNCTIONS=<symbol>` works too).
3. Add references to `<symbols>` to a source file that you know will be included.(for example the file where you `main` functions lives).
e.g.

```
// References to dummy symbols in order to force the inclusion of embind static constructors.
void* dummyRefs[] = [
   symbol1,
   symbol2,  // one symbol for each object file.
];
```

**Comment by brendandahl** [comment]
Are your bindings not working at runtime?
The output JS file won't have any of your bindings. They're all dynamically generated during static initialization at runtime.
